README.md

```diff
@@ -7,3 +7,522 @@ This repository contains submodules of all the repositories used for skywire dep
+
+## Runtime Dependencies by Service
+
+### Redis
+* address-resolver
+* network-monitor
+* dmsg-discovery
+
+### Postgres
+* transport-discovery
+* route-finder
+* service-discovery
+
+#### DB Setup
+
+Some services use postgresql.
+
+Run postgresql, make a database with UTF-8 character-set
+
+Pass the database, user, and password as envs
+
+```
+export PG_USER=username
+export PG_PASSWORD=pass
+export PG_DATABASE=sampledb
+```
+
+Specify the postgres host and port via flags
+
+```
+route-finder --pg-host localhost --pg-port 5432
+```
+
+All tables created automatically.
+
+
+## Required Services
+
+* address-resolver
+* network-monitor
+* dmsg-discovery
+* dmsg-server
+* transport-discovery
+* route-finder
+* service-discovery
+* setup-node
+
+A list of endpoints corresponding to some of these services in the current deployment
+
+* sd.skycoin.com/api/services?type=proxy
+* ar.skywire.skycoin.com/transports
+* tpd.skywire.skycoin.com/all-transports
+* dmsgd.skywire.skycoin.com/dmsg-discovery/entries
+* rf.skywire.skycoin.com/
+
+
+## Key generation
+
+Public/secret keypairs for all services have identical format can can be used interchangeably. There are multiple ways of potentially generating these, but for the sake of brevity the following command is suggested
+
+```
+skywire-cli config gen -n | head -n4 | tail -n2 | tee dmsgd-config.json
+grep -m1 "sk" dmsgd-config.json | cut -d '"' -f4 | tee -a dmsgd-config.json
+```
+
+those services which accept a secret key can then be run as follows
+```
+dmsg-discovery --sk $(tail -n1 dmsgd-config.json)
+```
+
+### Visor Config Bootstrap Endpoint
+
+The following endpoint is queried by the visor on config gen:
+https://conf.skywire.skycoin.com/
+
+This endpoint contains json which will become part of the visor's config
+```
+{
+  "dmsg_discovery": "http://dmsgd.skywire.skycoin.com",
+  "transport_discovery": "http://tpd.skywire.skycoin.com",
+  "address_resolver": "http://ar.skywire.skycoin.com",
+  "route_finder": "http://rf.skywire.skycoin.com",
+  "setup_nodes": [
+    "0324579f003e6b4048bae2def4365e634d8e0e3054a20fc7af49daf2a179658557"
+  ],
+  "uptime_tracker": "http://ut.skywire.skycoin.com",
+  "service_discovery": "http://sd.skycoin.com",
+  "stun_servers": [
+    "139.162.12.30:3478",
+    "170.187.228.181:3478",
+    "172.104.161.184:3478",
+    "170.187.231.137:3478",
+    "143.42.74.91:3478",
+    "170.187.225.78:3478",
+    "143.42.78.123:3478",
+    "139.162.12.244:3478"
+  ],
+  "dns_server": "1.1.1.1"
+}
+```
+
+This endpoint is used to avoid relying on hardcoded defaults for the production deployment endpoints in the visor. Without it, any changes here would require an updated version of skywire to manifest correct config default values
+
+A deployment of _all services as a set_ will require this for visors to query this endpoint during `config gen`
+
+### `dmsg-discovery`
+Run the Dmsg Discovery server
```

Comment:
I know you mentioned it before. But each of these services we list below should have an explicit reference to `make build`. Otherwise the adminsitrator/reader will run below command and its not going to work.

Reply:
Additionally, we need to tell them to `cd` to the correct directory before this is going to work.

Reply:
I sincerely doubt that they will use any build commands verbatim.

The compilation instructions exist as a guide or general reference.

`make` is not a required compilation dependency, doing `make build` is just there for convenience.

Technically, for versioning to be done correctly you need `make` and `git`.

The below commands are for running, I think the section for running any particular service should not be complicated by the compilation instructions which are _basically the same for any repo._

However, I will see if there is anything I can do to improve it.